README.md

```diff
@@ -1 +1,130 @@
-# earthstar-fetch
\ No newline at end of file
+# earthstar-fetch
+
+A `fetch()` interface for talking to EarthStar
+
+## Goals
+
+- Stick to existing conventions in HTTP-land (body, methods, prefer standard headers)
+- GET data from earthstar workspaces
+- GET an author address for a given name
+- PUT data into a workspace
+- Auto-generate identity keys from a 4 character name
+- Search via querystring params and path prefixes
+
+## Dream API
+
+Basic URL format:
+
+```
+earthstar://+gardening.xxxxxxxxx/path/to/key.json
+```
+
+GET document
+
+```JavaScript
+const res = await fetch('earthstar://+gardening.xxxxx/example')
+
+// Get the timestamp for when it was last updated
+res.headers.get('mtime') === "1231231231231"
+// Get the author for this document
+res.headers.get('earthstar-author') === '@bbbb.xxxxxxxx'
+// Get the signature for this document, in case you want it 🤷
+res.headers.get('earthstar-signature') === 'xxxxxxx'
+
+// Get the content of the document back out.
+// If it's JSON, you should use `.json()`
+const content = await res.text()
+```
+
+GET an identity 'name' - Generate an identity
+
+```JavaScript
+// Specify `@XXXX` where `X` is a lowercase ascii character
+// This will generate and store a keypair for this identity
+const res = fetch('earthstar://@xmpl')
+
+// Get the generated address for your identity
+const {address} = await res.json()
+
+address === '@xmpl.xxxxxxx'
+```
+
+POST some data, sign with identity
+
+```JavaScript
+// It's good practice to use a file extension so that the proper mime type gets set
+const res = await fetch('earthstar://+gardening.xxxxx/example.json', {
+  method: 'post',
+  body: JSON.stringify({
```

Comment:
Note that document content must be valid `utf-8`, no binary data.   (I want to change that in future versions of Earthstar)

Reply:
Good point! I'll add a note.